### PR TITLE
Image loading by diff / stateProvince cleanup / clustering index changes

### PR DESCRIPTION
**Pull request by djtfmartin**
PR includes:
* Cleanup of `stateProvince` values using a dictionary and GBIF's dictionary API
* Loading images and metadata updates using a diff of what is currently stored in image-service (to speed up image loading for large datasets)
* Command line tool for pruning of data directories of datasets that have been removed from the collectory
* Setting of licence using collectory information where record doesn't have a licence
* Changes to use `taxonRank` instead of `rank` to conform to DwC
* Changes to use `taxonRankID` instead of `rankID` for consistency
* Clustering changes to only associate a duplicate record with one representative record (includes end to end tests)
* Removal of `MetadataTransform` which didn't provide any indexable information for ALA's purposes
* Setting record level `licence` from Collectory value where record level licence not supplied
 

Includes:
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/281
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/271
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/306
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/299
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/278
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/307
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/258
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/239
https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/312

### CODE REVIEW

File: livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java [hunk 1]
```diff
@@ -30,4 +30,22 @@
...
 
   void setRecognisedPaths(String recognisedPaths);
+
+  @Description("Output a list of deletes")
+  @Default.Boolean(true)
+  boolean isOutputDeletes();
+
+  void setOutputDeletes(boolean outputDeletes);
+
+  @Description("Output a list of deletes")
```
Comment (charvolant) [inline livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java:40]: Description needs correction

File: livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java [hunk 2]
```diff
@@ -30,4 +30,22 @@
...
+  @Default.Boolean(true)
+  boolean isUploadNewImages();
+
+  void setUploadNewImages(boolean uploadNewImages);
+
+  @Description("Output a list of deletes")
```
Comment (charvolant) [inline livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java:46]: Description needs correction

**Review by charvolant — CHANGES_REQUESTED** [review]
Only a couple of minor things where an annotation was copied/pasted

File: livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java (hunk 2 above)
Comment (djtfmartin) [reply livingatlas/pipelines/src/main/java/au/org/ala/pipelines/options/ImageServicePipelineOptions.java]: fixed

**Pull request merged**